source/corvusoft/restbed/service.cpp

```diff
@@ -176,29 +176,32 @@ namespace restbed
         
         if ( m_pimpl->m_ready_handler not_eq nullptr )
         {
-            m_pimpl->m_io_service->post( m_pimpl->m_ready_handler );
+            m_pimpl->get_io_service()->post( m_pimpl->m_ready_handler );
         }
         
         m_pimpl->m_uptime = steady_clock::now( );
         unsigned int limit = m_pimpl->m_settings->get_worker_limit( );
         
-        if ( limit > 0 )
+        if ( m_pimpl->m_external_io_service == nullptr )
```

Comment:
This allows me to host different Asio based servers in a single io_service - TCP, UDP, WebSockets, HTTP/HTTPS. If you are interesting you may look into my CppServer project - https://github.com/chronoxor/CppServer/tree/master/source/server/asio